contracts/lib/AmountDeriver.sol

```diff
@@ -53,6 +51,9 @@ contract AmountDeriver is AmountDerivationErrors {
                 }
             }
 
+            // Derive time remaining until order expires
+            uint256 remaining = duration - elapsed;
```

Comment:
TODO: could this potentially be `unchecked`? Still need to investigate

Reply:
Constraints that are true from c4 review:

1. $startTime \ge block.timestamp_t$
2. $endTime > block.timestamp_t$
3. $endTime > startTime$ (1, 2)
4. $duration = endTime - startTime$
5. $duration > 0$ (3, 4)
6. $elapsed = block.timestamp_w - startTime$
7. $elapsed \ge 0$
8. $block.timestamp = endTime \implies invalid(order)$
9. $elapsed < duration$ (6, 8)

From this I think `unchecked` is fine.